### ISSUE TITLE
how does it takes care of cases when user has already rated the app ..?

### ISSUE THREAD

**Issue by ashishsajwan** [pushandplay/cordova-plugin-apprate#80]

**Comment by Mojo90** [comment]
same question. could it be that user gets a further prompt? should not happen. In my opinion if user rates app via prompt the user should not get any other popup regarding rating (except promptAgainForEachNewVersion is set to true - but is not my case). But a user recently reported that he rated the app via prompt and got a further prompt

**Comment by Wang0620** [comment]
I have same problem. 
I need to check if the user has already rated to the app. And if the user already did it, the app should not show rate dialog never. 
Could you please give me answer ASAP? 
Thanks.

**Comment by westonganger** [comment]
I would like this feature but I don't think it can be added. The major problem is that I don't think there is a way to determine if the user has actually rated the app or if they went into the store and never ended up rating it.

**Comment by ColDrekken** [comment]
Yes, i agree too, this should be a essential thing. Otherwise you have to set the usesUntilPrompt much higher to not bother your users, but that should not be the goal.

**Comment by ashishsajwan** [comment]
here is what i did. 

the moment user click on yes rate it now button (at this point user is redirected to app store/ play store) , i store a value in localstorage.  `isRated : True` 
& and before showing the rate popup i check for this value 
`if(isRated not set)  { 
   showPopup()
 } `

there is no certain way to check with play store if user has actually rated app or not. so we only rely on what option he has chosen when asked to rate the app. if he clicked on yes rate now then from here on don't ask for rating again. if he chose remind me late to cancel then don't set anything in localStorage.

**Comment by senior-engineer** [comment]
If we follow your way, until the users uninstall and reinstall the app, the users can not rate to the new released available version that the user want to rate.

**Comment by ashishsajwan** [comment]
@senior-engineer 
it should be very straight forward to implement this as well. 
in the first look i see the solution should be around saving the app version along with the isRated key. or even keeping the key dynamic. like `isRated-1.1 : true` then next version `isRated-1.2 : true`

now instead of looking for isRated Key we should look for `isRated-{currentVersion}`

**Comment by senior-engineer** [comment]
@ashishsajwan
I agree your idea about it. 

But the app can not know if the user has really rated to the app, although the user went to Apple Store and Google Play. Actually the user mightn't rate to the app on the App store screen. 

I have researched to find the solution to check if the user has already rated to the app, and seems there is a third party SDK to get it. 
But I did not use and implement it yet, so I do not know if it will work exactly.

**Comment by b1ackb0x** [comment]
My take on this: I would say that it would be a good to have option but it might violate user privacy. Check this answer on SO : http://stackoverflow.com/a/11284580/895666 and this: http://stackoverflow.com/a/5318824/895666 .The best way to deal with this is to leave it to the user to rate/review it. Once the user clicks on the **Rate Now** button it is anyways out of the app's purview. At the max, I would suggest you to use google analytics to at least track the clicks on **Rate Now** vs the actual count of ratings given by users.

How to deal with solution suggested? Track the button clicks based on callback method and button Ids. The way that I handle it is to track the clicked button and store it in local storage. The next time you need to show the rate dialog, check the last clicked button by getting the stored value first.
